tensorboard/plugins/profile/profile_plugin.py

```diff
@@ -142,65 +144,77 @@ def index_impl(self):
     for a specific tool "x" will have a suffix name TOOLS["x"].
     Example:
       log/
-        run1/
+        model1/
           plugins/
             profile/
-              host1.trace
-              host2.trace
-        run2/
+              run1/
+                host1.trace
+              run2/
+                host2.trace
+        model2/
           plugins/
             profile/
-              host1.trace
-              host2.trace
+              run1/
+                host1.trace
+              run2/
+                host2.trace
 
     Returns:
       A map from runs to tool names e.g.
-        {"run1": ["trace_viewer"], "run2": ["trace_viewer"]} for the example.
+        {"model1,run1": ["trace_viewer"], "model1,run2": ["trace_viewer"]}
+        for the example.
     """
-    # TODO(ioeric): use the following structure and use EventMultiplexer so that
-    # the plugin still works when logdir is set to root_logdir/run1/
-    #     root_logdir/
-    #       run1/
-    #         plugins/
-    #           profile/
-    #             host1.trace
-    #       run2/
-    #         plugins/
-    #           profile/
-    #             host2.trace
+
+    if any(self.run_to_tools.values()):
+      return self.run_to_tools
     run_to_tools = {}
-    if not tf.gfile.IsDirectory(self.plugin_logdir):
+    profile_logdirs = []
+    if not tf.gfile.IsDirectory(self.logdir):
       return run_to_tools
+    matcher = re.compile(r'profile')
+    folders = [self.logdir]
+    while folders:
+      cpath = folders.pop()
+      for fname in tf.gfile.ListDirectory(cpath):
+        fpath = os.path.join(cpath, fname)
+        if matcher.match(fname):
```

Comment:
Please check tf.gfile.IsDirectory(fpath) as well. Otherwise it will crash on line 187.

Reply:
Done.